packages/teams-js/test/public/calendar.spec.ts

```diff
@@ -0,0 +1,275 @@
+import { GlobalVars } from '../../src/internal/globalVars';
+import { FrameContexts } from '../../src/public';
+import { app } from '../../src/public/app';
+import { calendar } from '../../src/public/calendar';
+import { Utils } from '../utils';
+
+describe('calendar', () => {
+  // Use to send a mock message from the app.
+  const utils = new Utils();
+
+  beforeEach(() => {
+    utils.processMessage = null;
+    utils.messages = [];
+    utils.childMessages = [];
+    utils.childWindow.closed = false;
+    GlobalVars.frameContext = undefined;
+
+    // Set a mock window for testing
+    app._initialize(utils.mockWindow);
+  });
+
+  afterEach(() => {
+    // Reset the object since it's a singleton
+    if (app._uninitialize) {
+      app._uninitialize();
+    }
+  });
+
+  describe('openCalendarItem', () => {
+    const openCalendarItemParams: calendar.OpenCalendarItemParams = {
+      itemId: '1',
+    };
+
+    it('should not allow calls before initialization', async () => {
+      expect.assertions(1);
+
+      await calendar
+        .openCalendarItem(openCalendarItemParams)
+        .catch(e => expect(e).toMatchObject(new Error('The library has not yet been initialized')));
+    });
+
+    Object.keys(FrameContexts)
+      .map(k => FrameContexts[k])
+      .forEach(frameContext => {
+        it(`should not allow calls from ${frameContext} context`, async () => {
+          if (frameContext === FrameContexts.content) {
+            return;
+          }
+
+          expect.assertions(1);
+
+          await utils.initializeWithContext(frameContext);
+
+          await calendar
+            .openCalendarItem(openCalendarItemParams)
+            .catch(e =>
+              expect(e).toMatchObject(
+                new Error(
+                  `This call is only allowed in following contexts: ["content"]. Current context: "${frameContext}".`,
+                ),
+              ),
+            );
+        });
+      });
+
+    it('should not allow calls if runtime does not support calendar', async () => {
+      expect.assertions(1);
+
+      await utils.initializeWithContext('content');
+      utils.setRuntimeConfig({ apiVersion: 1, supports: {} });
+
+      await calendar.openCalendarItem(openCalendarItemParams).catch(e => expect(e).toBe('Not Supported'));
+    });
+
+    it('should throw if a null itemId is supplied', async () => {
+      expect.assertions(1);
+
+      await utils.initializeWithContext('content');
+      utils.setRuntimeConfig({ apiVersion: 1, supports: { calendar: {} } });
+
+      await calendar
+        .openCalendarItem({ itemId: null })
```

Comment:
Why doesn't this trigger compiler error 2322? I'm guessing the answer is because we don't have it turned on and we should probably have a bug to do that (or figure out what set of warnings it is in that we don't have enabled)

<img width="415" alt="image" src="https://user-images.githubusercontent.com/36546967/149600546-d00fcd1e-5529-4d3e-bbd6-3ea18eb28f43.png">

Reply:
@erinha Where should I parent a bug like this? Do you have a likely candidate in mind?

Reply:
Answered in chat

Reply:
bug has been opened